src/python/pants/help/help_printer.py

```diff
@@ -56,7 +57,17 @@ def print_hint() -> None:
         elif isinstance(self._help_request, OptionsHelp):
             self._print_options_help()
         elif isinstance(self._help_request, UnknownGoalHelp):
-            print("Unknown goals: {}".format(", ".join(self._help_request.unknown_goals)))
+            # Only print help and suggestions for the first unknown goal.
+            # It gets confusing to try and show suggestions for multiple cases.
```

Comment:
Good move. The user can recursively whittle down unknown goals until the whole command line makes sense.